src/txn/txn_recover.c

```diff
@@ -821,7 +821,7 @@ __recovery_file_scan(WT_RECOVERY *r)
      * Set the connection level file id tracker, as such upon creation of a new file we'll begin
      * from the latest file id.
      */
-    S2C(r->session)->next_file_id = r->max_fileid;
+    S2C(r->session)->next_file_id = WT_BTREE_ID_UNNAMESPACED(r->max_fileid);
```

Comment:
Should we just assign it to r->max_fileid for the upgrade case? We don't want to maintain backward compatibility?

Reply:
That's a bigger design choice - doing that is effectively saying that the max file id is stored as the namespaced version, which is fine, but it means instead of incrementing it we'd have to add three each time, among other annoying but solvable things.

Reply:
Yes, the final decision can be made later. My suggestion is to add a comment here in case we forget about it.

Reply:
Fair call - done in c83e0f6.